### DISCUSSION TITLE
How do I uninstall the project?

### DISCUSSION THREAD

**Discussion by cedewey** [filips123/PWAsForFirefox#37] [Q&A] [answered]
I've uninstalled the Firefox Add-On, but how do I also uninstall the command line tool (I'm using the Debian version) and subsequent desktop PWAs?

**Comment by filips123** [comment, marked as answer]
You need to do these steps:
1. Uninstall PWAs from the addon (open PWA list, then remove each PWA) or using the command-line program (`firefoxpwa profile list` to list all PWAs, then `firefoxpwa site uninstall ID` to uninstall each PWA). This will remove PWAs from the application list, but it will still keep their data.
2. Uninstall the DEB package (`sudo dpkg -r firefoxpwa`). This will still keep data.
3. Remove all data by deleting `/usr/share/firefoxpwa` and `~/.local/share/firefoxpwa`.

Please let me know if you have any problem.

**Reply by juzijun233** [reply]
how to do it on macOS?

**Reply by filips123** [reply]
Package can be uninstalled with brew. Data are stored in `$HOME/Library/Application Support/firefoxpwa/`, which you can remove if you want.

**Reply by Superpigdots** [reply]
Can you please explain this to me like I am 10? I find your instructions unable to be followed. I want to uninstall the PWAs extension and any related saved data from my browser/PC and also restore any altered settings. I already see that your directions do not account for added C++ installs, that one other program install, and the altered Firefox about:config setting.

1. In step one, are you saying to uninstall PWAs extension in Firefox? Or are you referring to uninstalling a PWA that I have added to my PC? What do you mean by "uninstall PWAs from the addon"? What addon?
2. In step two, what is a DEB package and where/how do I uninstall it?
3. Where do I delete said paths in step 3? How?

Your installation package/directions are super detailed and fool-proof. Could you please do the same for uninstall directions? Maybe adding it to the Github repository documentation would be useful too.

**Reply by filips123** [reply]
1. You should first uninstall PWAs and their profiles using the PWAsForFirefox extension that you've used to install them. Then, you should uninstall the extension itself.

2. A DEB package is specific to Linux. On Windows, you should just uninstall the PWAsForFirefox program from the Windows Settings or Control Panel, like any other program (after uninstalling the PWAs as mentioned in the first step).

3. You should delete [the user data directory](https://pwasforfirefox.filips.si/resources/installation-directories/#user-data).

You probably shouldn't delete Visual C++ Redistributable, as it may be used by other programs.

I will add the instructions to the documentation soon.

**Reply by Superpigdots** [reply]
1. Okay, so in the PWAs extension interface, I was able to delete the PWA I installed. It said I was also able to delete the default profile I was using, but it just immediately shows still and never goes away. Do I need to do anything in my Firefox files on my PC in order to delete the other profile info entirely? I was also able to uninstall the extension.
2. I was able to delete the program from my PC.
3. I found and deleted the folder `%appdata%\FirefoxPWA\`. Is that all I have to do? Just simply right click>delete it?

So, is that all there is to it?

**Reply by filips123** [reply]
You can't "completely remove" the default profile from the extension, as it's a base default profile. However, it's stored in `%appdata%\FirefoxPWA\`, so once you remove that folder, the default profile will also be deleted. So, what you've done should be enough to complely remove PWAsForFirefox and all data.

**Reply by Superpigdots** [reply]
Okay, thank you so much! Should I also go into the Firefox about:config and turn off thst setting I changed in the PWAs setup?

**Reply by filips123** [reply]
That isn't necessary, as those settings were applied to the web app profiles, which you've already deleted.

**Reply by Superpigdots** [reply]
Thank you!

**Comment by thekarananand** [comment]
How to remove the repo/copr of FirefoxPWA on Fedora?

**Reply by filips123** [reply]
Instructions for this are available on [Fedora docs](https://docs.fedoraproject.org/en-US/quick-docs/adding-or-removing-software-repositories-in-fedora/#_removing_repositories). PWAsForFirefox's repo is named `firefoxpwa.repo`.
